ledger/src/blockstore_processor.rs

```diff
@@ -1171,6 +1196,26 @@ impl TransactionStatusSender {
     }
 }
 
+pub type CacheBlockTimeSender = Sender<Arc<Bank>>;
+
+pub fn cache_block_time(
+    blockstore: &Blockstore,
+    bank: &Arc<Bank>,
+    cache_block_time_sender: Option<&CacheBlockTimeSender>,
+) {
+    if let Some(cache_block_time_sender) = cache_block_time_sender {
+        if blockstore
+            .get_block_time(bank.slot())
+            .unwrap_or_default()
+            .is_none()
```

Comment:
I think we can move this read into the background thread as well, maybe even directly in the `blockstore::cache_block_time()` method

Reply:
Done here https://github.com/solana-labs/solana/pull/17109/commits/45d5815f3b10b144fd55999fbdedfedcbc0aad25